iputil/ip.go

```diff
@@ -0,0 +1,70 @@
+// Package iputil defines type that will marshal/unmarshal an IP address as
+// human readable IPv4 string rather than an unreadable byte stream.
+package iputil
+
+import (
+	"bytes"
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+	"net"
+)
+
+// IP is a special wrapped type of net.IP
+type IP net.IP
+
+// Gt will return if the IP is greater than the given IP
+func (ip IP) Gt(start net.IP) bool {
+	return bytes.Compare(ip, start) >= 0
+}
+
+// Lt will return if the IP is less than the given IP
+func (ip IP) Lt(end net.IP) bool {
+	return bytes.Compare(ip, end) <= 0
+}
+
+// InRange will return if the IP is within the given range of addresses
+func (ip IP) InRange(start, end net.IP) bool {
+	return ip.Gt(start) && ip.Lt(end)
+}
+
+// String returns the IP as a string
+func (ip IP) String() string {
+	return net.IP(ip).String()
+}
+
+// Value will get the value for storing in SQL
+func (ip IP) Value() (driver.Value, error) {
+	return ip.String(), nil
+}
+
+// Scan will return the IP from the database
+func (ip *IP) Scan(value interface{}) error {
+	str, ok := value.([]byte)
+	if !ok {
+		return fmt.Errorf("iputil: cannot scan type %T into ip.IP: %v", value, value)
```

Comment:
The `Value()` returns a string, but this only accepts `[]byte`. You need to be careful with this, as the database driver can pass either, and this may even change between different versions of the DB driver. I discovered this the hard way a while ago when the SQLite driver changed it a while ago. That it passes `[]byte` here is really just an implementation detail of the MySQL driver and ideally it should accept both a `string` and `[]byte` with e.g. a type switch.

Reply:
Fair point, will check both then to be sure for safety. I expected a string when working on this but I got a byte slice.